### PR TITLE
Fix for issue #179

### LINKED ISSUE #179: Switching off network throws errors in the @ buzz app
- [x] Multiple connections are being created

After switching on the network back, Multiple connections are being created and server is exiting with the following error.

Server Logs

FINER|2021-06-28 17:55:45.526536|AtSecondaryServer|In _listen - clientSocket.peerCertificate : null 

FINER|2021-06-28 17:55:45.612642|AtSecondaryServer|In _listen - clientSocket.peerCertificate : null 

FINER|2021-06-28 17:55:45.613868|AtSecondaryServer|In _listen - clientSocket.peerCertificate : null 

FINER|2021-06-28 17:55:45.614955|AtSecondaryServer|In _listen - clientSocket.peerCertificate : null 

FINER|2021-06-28 17:55:46.182049|AtSecondaryServer|In _listen - clientSocket.peerCertificate : null 

FINER|2021-06-28 17:55:46.422351|AtSecondaryServer|In _listen - clientSocket.peerCertificate : null 

FINER|2021-06-28 17:55:46.424437|AtSecondaryServer|In _listen - clientSocket.peerCertificate : null 

FINER|2021-06-28 17:55:46.502695|AtSecondaryServer|In _listen - clientSocket.peerCertificate : null 

Exhausted heap space, trying to allocate 64 bytes.
Unhandled exception:
Out of Memory
#0      Object._instanceOf (dart:core-patch/object_patch.dart:71:33)
#1      _FutureListener.shouldChain (dart:async/future_impl.dart:182:52)
#2      Future._propagateToListeners (dart:async/future_impl.dart:747:22)
#3      Future._complete (dart:async/future_impl.dart:530:7)
#4      new Future.delayed.<anonymous closure> (dart:async/future.dart:312:16)
#5      Timer._createTimer.<anonymous closure> (dart:async-patch/timer_patch.dart:18:15)
#6      _Timer._runTimers (dart:isolate-patch/timer_impl.dart:395:19)
#7      _Timer._handleMessage (dart:isolate-patch/timer_impl.dart:426:5)

- [x] Network Error in at_buzz

**Describe the bug**
Switching off the network in the mobile throws error

**To Reproduce**
1) Switch off the network in Device B
2) Share a persona from Device A to Device B
3) Switch on the network on Device B. I got the notifications. But the data which was previously shared was not seen and I got this issue in the app logs

App Logs

[ERROR:flutter/lib/ui/ui_dart_state.cc(186)] Unhandled Exception: LateInitializationError: Local 'result' has not been initialized.
E/flutter (22125): #0 LateError._throwLocalNotInitialized (dart:_internal-patch/internal_patch.dart:208:5)
E/flutter (22125): #1 Monitor._getQueueResponse
package:at_client/…/manager/monitor.dart:200
E/flutter (22125): <asynchronous suspension>
E/flutter (22125): #2 Monitor._authenticateConnection
package:at_client/…/manager/monitor.dart:134
E/flutter (22125): <asynchronous suspension>
E/flutter (22125): #3 Monitor.start
package:at_client/…/manager/monitor.dart:123
E/flutter (22125): <asynchronous suspension>

SEVERE|2021-06-22 14:04:22.319895|AtLookup|Error in remote verb execution Instance of 'UnAuthenticatedException'
E/flutter (25807): [ERROR:flutter/lib/ui/ui_dart_state.cc(186)] Unhandled Exception: NoSuchMethodError: Closure call with mismatched arguments: function 'SyncManager._handleError'
E/flutter (25807): Receiver: Closure: (dynamic, Exception, Function, Function) => void from Function '_handleError@235377768':.
E/flutter (25807): Tried calling: SyncManager._handleError(Instance of 'SyncManager', Instance of 'AtLookUpException')
E/flutter (25807): Found: SyncManager._handleError(dynamic, Exception, Function, Function) => void
E/flutter (25807): #0 Object.noSuchMethod (dart:core-patch/object_patch.dart:54:5)
E/flutter (25807): #1 SyncManager._handleError
package:at_client/…/manager/sync_manager.dart:62
E/flutter (25807): #2 SyncManager.syncOnce
package:at_client/…/manager/sync_manager.dart:114
E/flutter (25807): <asynchronous suspension>
E/flutter (25807): #3 SyncManager.sync
package:at_client/…/manager/sync_manager.dart:52
E/flutter (25807): <asynchronous suspension>
E/flutter (25807): #4 AtBuzzClientService.refresh
package:Buzz_App/…/services/AtBuzzClientService.dart:71
E/flutter (25807): <asynchronous suspension>
E/flutter (25807): #5 _SharedDetailsState._onRefresh
package:Buzz_App/…/sharing/shared_data.dart:384
E/flutter (25807): <asynchronous suspension>

Server Logs

FINER|2021-06-22 14:07:26.269664|FromVerbHandler|Verb : from  Response: Response{_data: data:_946b2c63-e4b4-4367-9ad7-209636e5cf63@ alice:74cdc7d7-7571-4ff0-b4ed-922744ba8b0e, _type: null, _isError: false, _errorMessage: null} 

FINER|2021-06-22 14:07:26.269761|FromResponseHandler|Got response: Response{_data: data:_946b2c63-e4b4-4367-9ad7-209636e5cf63@ alice:74cdc7d7-7571-4ff0-b4ed-922744ba8b0e, _type: null, _isError: false, _errorMessage: null} 

FINER|2021-06-22 14:07:26.331773|InboundListener|command received: pkam:gspAD3sU9qfl6xdDNb1Ny4QmV7Hip87gUybaRo6MTVti3EzHWxZQINp+BxElYx688+zQIz5xUxxB6wBb1NFqwdM+D9U2A1DEf0f2pMIzGUAZcdjX7Ev1rRlHWIynAw3edldJ22uX5E6qjpbc5zvz/VNwe7298JgbFf6UH1nSaNDHedqqyZVWeesS/2Pc4YYSvVWSqPCukLbNNuXuFDvehWtSSJZPwmOGYayJTc5dHIsEqEXckvBqLxFna0Al7oyBaCKF1cJnpS1hZnJhSIIy0sbJy09t9aa7142NHHIiRynNZuGBL+2gjzRYX6kf0G/Tdj8DcEcttOvUAQk4jzzv2A== sessionID:_946b2c63-e4b4-4367-9ad7-209636e5cf63 

FINER|2021-06-22 14:07:26.331932|AtSecondaryServer|inside _executeVerbCallBack: pkam:gspAD3sU9qfl6xdDNb1Ny4QmV7Hip87gUybaRo6MTVti3EzHWxZQINp+BxElYx688+zQIz5xUxxB6wBb1NFqwdM+D9U2A1DEf0f2pMIzGUAZcdjX7Ev1rRlHWIynAw3edldJ22uX5E6qjpbc5zvz/VNwe7298JgbFf6UH1nSaNDHedqqyZVWeesS/2Pc4YYSvVWSqPCukLbNNuXuFDvehWtSSJZPwmOGYayJTc5dHIsEqEXckvBqLxFna0Al7oyBaCKF1cJnpS1hZnJhSIIy0sbJy09t9aa7142NHHIiRynNZuGBL+2gjzRYX6kf0G/Tdj8DcEcttOvUAQk4jzzv2A== 

FINER|2021-06-22 14:07:26.332102|DefaultVerbExecutor|verb handler found : PkamVerbHandler 

FINER|2021-06-22 14:07:26.334661|PkamVerbHandler|pkam auth:true 

FINER|2021-06-22 14:07:26.334865|PkamVerbHandler|Verb : pkam  Response: Response{_data: success, _type: null, _isError: false, _errorMessage: null} 

FINER|2021-06-22 14:07:26.335303|DefaultResponseHandler|Got response: Response{_data: success, _type: null, _isError: false, _errorMessage: null} 

FINER|2021-06-22 14:07:26.425731|InboundListener|command received: monitor sessionID:_946b2c63-e4b4-4367-9ad7-209636e5cf63 

FINER|2021-06-22 14:07:26.425805|AtSecondaryServer|inside _executeVerbCallBack: monitor 

FINER|2021-06-22 14:07:26.425968|DefaultVerbExecutor|verb handler found : MonitorVerbHandler 

FINER|2021-06-22 14:07:26.426141|MonitorVerbHandler|Verb : monitor  Response: Response{_data: null, _type: null, _isError: false, _errorMessage: null} 

FINER|2021-06-22 14:07:26.426205|MonitorResponseHandler|Got response: Response{_data: null, _type: null, _isError: false, _errorMessage: null} 

SEVERE|2021-06-22 14:07:36.691085|AtSecondaryServer|HandshakeException: Connection terminated during handshake 

SEVERE|2021-06-22 14:07:36.695765|AtSecondaryServer|HandshakeException: Connection terminated during handshake 

SEVERE|2021-06-22 14:07:36.701979|AtSecondaryServer|HandshakeException: Connection terminated during handshake 

FINER|2021-06-22 14:07:37.014835|AtSecondaryServer|In _listen - clientSocket.peerCertificate : null 

FINER|2021-06-22 14:07:37.020847|AtSecondaryServer|In _listen - clientSocket.peerCertificate : null 

FINER|2021-06-22 14:07:37.021907|AtSecondaryServer|In _listen - clientSocket.peerCertificate : null 

FINER|2021-06-22 14:07:37.053252|InboundListener|command received: from:@ alice sessionID:_80c42e57-f22c-405d-97b4-bc61cf813de8 

FINER|2021-06-22 14:07:37.053519|AtSecondaryServer|inside _executeVerbCallBack: from:@ alice 

FINER|2021-06-22 14:07:37.053624|DefaultVerbExecutor|verb handler found : FromVerbHandler 

FINER|2021-06-22 14:07:37.054989|FromVerbHandler|fromAtSign : @ alice currentAtSign : @ alice 

FINER|2021-06-22 14:07:37.056212|FromVerbHandler|Verb : from  Response: Response{_data: data:_80c42e57-f22c-405d-97b4-bc61cf813de8@ alice:eefeeffd-4398-4b22-b4d9-ba68698ff97c, _type: null, _isError: false, _errorMessage: null} 

FINER|2021-06-22 14:07:37.056355|FromResponseHandler|Got response: Response{_data: data:_80c42e57-f22c-405d-97b4-bc61cf813de8@ alice:eefeeffd-4398-4b22-b4d9-ba68698ff97c, _type: null, _isError: false, _errorMessage: null} 

FINER|2021-06-22 14:07:37.059548|InboundListener|command received: from:@ alice sessionID:_12faccdd-a70f-4f64-9dbb-8b85d3325e12 

FINER|2021-06-22 14:07:37.059642|AtSecondaryServer|inside _executeVerbCallBack: from:@ alice 

FINER|2021-06-22 14:07:37.059721|DefaultVerbExecutor|verb handler found : FromVerbHandler 

FINER|2021-06-22 14:07:37.060491|FromVerbHandler|fromAtSign : @ alice currentAtSign : @ alice 

FINER|2021-06-22 14:07:37.060998|FromVerbHandler|Verb : from  Response: Response{_data: data:_12faccdd-a70f-4f64-9dbb-8b85d3325e12@ alice:93912aed-5fb2-4166-9be9-1f8561d80a28, _type: null, _isError: false, _errorMessage: null} 

FINER|2021-06-22 14:07:37.061062|FromResponseHandler|Got response: Response{_data: data:_12faccdd-a70f-4f64-9dbb-8b85d3325e12@ alice:93912aed-5fb2-4166-9be9-1f8561d80a28, _type: null, _isError: false, _errorMessage: null} 

FINER|2021-06-22 14:07:37.061679|InboundListener|command received: from:@ alice sessionID:_b8ab6f53-85b1-4828-8644-9bc009753b83 

FINER|2021-06-22 14:07:37.061870|AtSecondaryServer|inside _executeVerbCallBack: from:@ alice 

FINER|2021-06-22 14:07:37.062040|DefaultVerbExecutor|verb handler found : FromVerbHandler 

FINER|2021-06-22 14:07:37.066733|FromVerbHandler|fromAtSign : @ alice currentAtSign : @ alice 

FINER|2021-06-22 14:07:37.067037|FromVerbHandler|Verb : from  Response: Response{_data: data:_b8ab6f53-85b1-4828-8644-9bc009753b83@ alice:39621307-5ae1-4561-b3b7-14fbc803cc89, _type: null, _isError: false, _errorMessage: null} 

FINER|2021-06-22 14:07:37.067638|FromResponseHandler|Got response: Response{_data: data:_b8ab6f53-85b1-4828-8644-9bc009753b83@ alice:39621307-5ae1-4561-b3b7-14fbc803cc89, _type: null, _isError: false, _errorMessage: null} 

FINER|2021-06-22 14:07:37.229841|InboundListener|command received: pkam:TH4mD6+LZFC6WaugKKjmBB7xavzy1c3EW97vCukdpyGZYRbjWLKqRq3/Sxt3btT+jujnu/yHOTBhRb3xKkvv/BDLMuVOJIoQUO7oSjqcesmFMuXZpdzJIsUYae7wSCwbOtCiV7kuG4UbKbEBCpcYbZ4N4x3gXsY680KwSbCv9iEamTW0/8vowvnUM7LjnfbsjFRvR6a0y70lq//NmQSHQmNbz+zfE2pBAPZrWmFBFAlj0ncOXRhJGGlOMpfYb6HvI1xod1kw4793EOSu+VxflluuVb2onSywRrAXlCFqz5uApvBxxJh9qASMAjb4Fc6InhT61EnarfQV4oDS3lbi8Q== sessionID:_b8ab6f53-85b1-4828-8644-9bc009753b83 

FINER|2021-06-22 14:07:37.229924|AtSecondaryServer|inside _executeVerbCallBack: pkam:TH4mD6+LZFC6WaugKKjmBB7xavzy1c3EW97vCukdpyGZYRbjWLKqRq3/Sxt3btT+jujnu/yHOTBhRb3xKkvv/BDLMuVOJIoQUO7oSjqcesmFMuXZpdzJIsUYae7wSCwbOtCiV7kuG4UbKbEBCpcYbZ4N4x3gXsY680KwSbCv9iEamTW0/8vowvnUM7LjnfbsjFRvR6a0y70lq//NmQSHQmNbz+zfE2pBAPZrWmFBFAlj0ncOXRhJGGlOMpfYb6HvI1xod1kw4793EOSu+VxflluuVb2onSywRrAXlCFqz5uApvBxxJh9qASMAjb4Fc6InhT61EnarfQV4oDS3lbi8Q== 

FINER|2021-06-22 14:07:37.230056|DefaultVerbExecutor|verb handler found : PkamVerbHandler 

FINER|2021-06-22 14:07:37.231769|PkamVerbHandler|pkam auth:false 

SEVERE|2021-06-22 14:07:37.231824|PkamVerbHandler|pkam authentication failed 

#0      PkamVerbHandler.processVerb (package:at_secondary/src/verb/handler/pkam_verb_handler.dart:72:7)
<asynchronous suspension>
#1      AbstractVerbHandler.processInternal (package:at_secondary/src/verb/handler/abstract_verb_handler.dart:57:7)
<asynchronous suspension>
#2      AbstractVerbHandler.process (package:at_secondary/src/verb/handler/abstract_verb_handler.dart:32:20)
<asynchronous suspension>
#3      DefaultVerbExecutor.execute (package:at_secondary/src/verb/executor/default_verb_executor.dart:27:7)
<asynchronous suspension>
#4      AtSecondaryServerImpl._executeVerbCallBack (package:at_secondary/src/server/at_secondary_impl.dart:293:7)
<asynchronous suspension>
#5      InboundMessageListener._messageHandler (package:at_secondary/src/connection/inbound/inbound_message_listener.dart:59:9)
<asynchronous suspension>

SEVERE|2021-06-22 14:07:37.232089|AtSecondaryServer|Instance of 'UnAuthenticatedException' 

SEVERE|2021-06-22 14:07:38.511715|AtSecondaryServer|HandshakeException: Connection terminated during handshake 

SEVERE|2021-06-22 14:07:38.696114|AtSecondaryServer|HandshakeException: Connection terminated during handshake 

FINER|2021-06-22 14:07:38.826006|AtSecondaryServer|In _listen - clientSocket.peerCertificate : null 

**Expected behavior**
Previously shared data and newly shared personas should be displayed without any issues after switching on the network.

**Screenshots**

**Smartphone (please complete the following information):**
 - Device: Google pixel
 - OS: Android

**Were you using an @application when the bug was found?**
  -  at_buzz

Branch - sdk_refactoring
app_branch - bugs_and_enhancements

**Additional context**

### PR DESCRIPTION
**Pull request by sitaram-kalluri**
**- What I did**  Fix git issue #179 

**- How I did it**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: at_client/lib/src/manager/monitor.dart [hunk 1]
```diff
@@ -214,7 +217,7 @@ class Monitor {
   /// Stops the monitor. Call [Monitor#start] to start it again.
   void stop() {
     status = MonitorStatus.Stopped;
-    _monitorConnection.close();
+    _monitorConnection.getSocket().destroy();
```
Comment (murali-shris) [inline at_client/lib/src/manager/monitor.dart]: connection.close() internally calls socket.destroy..so this change is not required

File: at_client/lib/src/manager/monitor.dart [hunk 2]
```diff
@@ -84,7 +84,7 @@ class Monitor {
     _regex = monitorPreference.regex;
     _keepAlive = monitorPreference.keepAlive;
     _lastNotificationTime = monitorPreference.lastNotificationTime;
-    _remoteSecondary = RemoteSecondary(atSign, preference);
+    _remoteSecondary ??= RemoteSecondary(atSign, preference);
```
Comment (murali-shris) [inline at_client/lib/src/manager/monitor.dart:87]: if monitor is created with active connection and then internet on the client side drops..the remotesecondary instance that was created initially will be obsolete..right?

File: at_client/lib/src/manager/monitor.dart (hunk 1 above)
Comment (sitaram-kalluri) [reply at_client/lib/src/manager/monitor.dart]: Reverted with monitor.close() method.

File: at_client/lib/src/manager/monitor.dart (hunk 2 above)
Comment (sitaram-kalluri) [reply at_client/lib/src/manager/monitor.dart:87]: Any command from remote secondary goes via the sendCommand method in the at_lookup_impl. The sendCommand method creates a new connection whenever a connection is not available.

**Review by murali-shris — APPROVED** [review]

**Pull request merged**